File: swift/217_contains_duplicate.swift
class Solution {
    func containsDuplicate(_ nums: [Int]) -> Bool {
        var hashMap: [Int: Int] = [:]
        
        for num in nums {
            if let _ = hashMap[num] {
                return true
            } else {
                hashMap[num] = 0
            }
        }
        
        return false
     
// another approach
//         let sortedNums = nums.sorted()
        
//         for i in 0..<(nums.count-1) {
//             if sortedNums[i] == sortedNums[i+1] {
//                 return true
//             }
//         }
        
//         return false
    }
}

File: python/217_contains_duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hashset = set()

        for num in nums:
            if num in hashset:
                return True
            else:
                hashset.add(num)
        
        return False
        
#         nums.sort()
        
#         for i in range(len(nums) - 1):
#             if nums[i] == nums[i+1]:
#                 return True
        
#         return False
        
